src/pages/Send/hooks/useSend/usePVMSend.ts

```diff
@@ -73,9 +175,32 @@ export const usePvmSend: SendAdapterPVM = ({
         return setErrorAndEndValidating(errorReason);
       }
 
+      if (typeof gasPrice === 'bigint' && feeState) {
+        if (feeState.price > gasPrice) {
+          return setErrorAndEndValidating(SendErrorMessage.INVALID_NETWORK_FEE);
+        } else if (gasPrice > feeState.price * 2n) {
+          return setErrorAndEndValidating(
+            SendErrorMessage.EXCESSIVE_NETWORK_FEE
+          );
+        }
+      }
+
+      if (provider.isEtnaEnabled() && !feeState) {
```

Comment:
I know `isEtnaEnabled` is from the provider but it would be nice to have a more generic solution like `isUpgradeEnabled(PlatformUpgrage.Etna)` or something. Doesn't have to be in the current PR though